### PR TITLE
Don't give up when missing the initial transmit window

### LINKED ISSUE #77: Don't give up when missing the initial transmit window
Currently the link layer will abort the connection when it does not receive the initial connect event:

https://github.com/jonas-schievink/rubble/blob/0c219f5fd1ef24cc6ca6f6e1edb02556e5fdd64f/rubble/src/link/connection.rs#L338-L345

This shouldn't happen. Instead, we should do what the comment suggests (and check it against the spec).

The effect of this bug is that establishing a connection can spuriously fail, so this is a reliability issue.

### PR DESCRIPTION
**Pull request by JOE1994**
Hello :crab: , this PR aims to close #77.

In the example from bluetooth specs (5.2) where the master misses the initial transmit window, the slave hops to a new channel for the next connection event. So I followed the example, to hop channels when the master device misses the initial transmit window.
image

Thank you for reviewing this PR ! :smile_cat:

### CODE REVIEW

**Review by jonas-schievink — APPROVED** [review]
Looks correct, thanks!

Did you manage to test this, by chance?

File: rubble/src/link/connection.rs [hunk 1]
```diff
@@ -57,6 +57,9 @@ pub struct Connection<C: Config> {
     /// Contains the *instant* at which it should be applied to the Link Layer state.
     update_data: Option<LlcpUpdate>,
 
+    /// 'transmitWindowSize' of current connection.
```
Comment (jonas-schievink) [inline rubble/src/link/connection.rs:60]:
```suggestion
    /// `transmitWindowSize` of current connection.
```

**Comment by JOE1994** [comment]
> Looks correct, thanks!
> 
> Did you manage to test this, by chance?

I haven't yet tested it on hardware,
but I can try testing once I receive the `nRF52840` board I ordered a few days ago :)

p.s.
I found a weird mistake I made in this PR:
I pushed the transmit window by `(connInterval - win_size)` and not by `connInterval`.. :shrug:
I'll add a follow-up fix commit right away :)

**Comment by JOE1994** [comment]
I made a force-push to squash the follow-up fix with the previous commit.

**Pull request closed without merging**